src/openMVG/matching/matcher_hnsw.hpp

```diff
@@ -51,10 +51,18 @@ class HNSWMatcher: public ArrayMatcher<Scalar, Metric>
     int dimension
   ) override
   {
+    HNSWmetric.reset(nullptr);
+    HNSWmatcher.reset(nullptr);
+    
     if (nbRows < 1)
-    {
-      HNSWmetric.reset(nullptr);
-      HNSWmatcher.reset(nullptr);  
+      return false;
+      
+    // Returns early if DistanceType is not supported.
+    // This avoids the case where HNSWmetric is null, which causes an error when 
+    // HNSWmatcher initializes.
+    if (
+        typeid(DistanceType) != typeid(int) &&
+        typeid(DistanceType) != typeid(float)) {
```

Comment:
maybe remove the else branch below and put here the
`std::cerr << "HNSW matcher: this type of distance is not handled Yet" << std::endl;`

Reply:
Cool, taken care of.

Still seems like there's more that could be done. What do you think of using explicit template instantiations for int and float? That would cause compile errors instead of this soft error, so it changes the interface. What do you all think?